runtime/src/main/java/org/corfudb/protocols/wireprotocol/StreamAddressSpace.java

```diff
@@ -0,0 +1,154 @@
+package org.corfudb.protocols.wireprotocol;
+
+import org.corfudb.runtime.view.Address;
+import org.roaringbitmap.longlong.LongIterator;
+import org.roaringbitmap.longlong.Roaring64NavigableMap;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import java.util.NavigableSet;
+import java.util.TreeSet;
+
+/**
+ * This class represents the space of all addresses belonging to a stream.
+ *
+ * A stream's address space is defined by:
+ *       1. The collection of all addresses that belong to this stream.
+ *       2. The trim mark (last trimmed address, i.e., an address that is no longer present and that was subsumed by
+ *       a checkpoint).
+  *
+ * Created by annym on 03/06/2019
+ */
+public class StreamAddressSpace {
+
+    private static final int NO_ADDRESSES = 0;
```

Comment:
why NO_ADDRESES does not equal to Address.NON_ADDRESS?
why not to use Address.NON_ADDRESS instead?